lxd/network/driver_bridge.go

```diff
@@ -2543,6 +2543,32 @@ func (n *bridge) getExternalSubnetInUse() ([]externalSubnetUsage, error) {
 	externalSubnets = append(externalSubnets, bridgeNetworkExternalSubnets...)
 	externalSubnets = append(externalSubnets, bridgedNICExternalRoutes...)
 
+	// Detect if there are any conflicting proxy devices on all instances with the to be created network forward
+	err = n.state.DB.Cluster.InstanceList(context.TODO(), func(inst db.InstanceArgs, p api.Project) error {
+		devices := db.ExpandInstanceDevices(inst.Devices, inst.Profiles)
+		for devName, devConfig := range devices {
+			if devConfig["type"] == "proxy" {
+				proxyListenAddr := strings.Split(devConfig["listen"], ":")[1]
```

Comment:
Stuff like this makes me uncomfortable that if there was ever a listen setting with no `:` in it then this would cause LXD to panic. Better to check the length of the slice first before using it.

Reply:
Suggest moving the `ProxyParseAddr` function into network/network_utils.go and then you can use it from here and in proxy device.